app/src/main/java/org/jellyfin/mobile/fragment/ConnectFragment.kt

```diff
@@ -157,61 +162,54 @@ class ConnectFragment : Fragment() {
         connectionErrorText.isVisible = false
     }
 
-    private suspend fun checkServerUrlAndConnection(enteredUrl: String): HttpUrl? {
+    private suspend fun checkServerUrlAndConnection(enteredUrl: String): String? {
         Timber.i("checkServerUrlAndConnection $enteredUrl")
 
-        val normalizedUrl = enteredUrl.run {
-            if (lastOrNull() == '/') this
-            else "$this/"
-        }
-        val urls = jellyfin.discovery.addressCandidates(normalizedUrl)
-        Timber.i("Address candidates are $urls")
-
-        var httpUrl: HttpUrl? = null
-        var serverInfo: PublicSystemInfo? = null
-        loop@ for (url in urls) {
-            httpUrl = url.toHttpUrlOrNull()
+        val candidates = jellyfin.discovery.getAddressCandidates(enteredUrl)
+        Timber.i("Address candidates are $candidates")
 
-            if (httpUrl == null) {
-                showConnectionError(R.string.connection_error_invalid_format)
-                return null // Format is invalid, don't try any other variants
-            }
+        val recommendedServer = jellyfin.discovery.getRecommendedServer(candidates, false)
 
-            // Set API client address
-            apiClient.ChangeServerLocation(httpUrl.toString().trimEnd('/'))
+        // No server found that replied
+        if (recommendedServer == null) {
+            Timber.i("No recommended server found")
 
-            serverInfo = apiClient.getPublicSystemInfo()
-            if (serverInfo != null)
-                break@loop
+            //TODO add candidates to error
+            showConnectionError(R.string.connection_error_cannot_connect)
+            return null
         }
 
-        if (httpUrl == null || serverInfo == null) {
-            Timber.w("Failed to find server info, url was $httpUrl")
+        val systemInfo = recommendedServer.systemInfo
 
-            showConnectionError()
+        // System Info is missing, shouldn't be able to happen but check just in case
+        if (systemInfo == null) {
+            Timber.w("Recommended server did not contain system information!")
+
+            showConnectionError(R.string.connection_error_invalid_version)
             return null
         }
 
-        val version = serverInfo.version
-            .split('.')
-            .mapNotNull(String::toIntOrNull)
+        val version = systemInfo.version?.let { ServerVersion.fromString(it) }
 
         val isValidInstance = when {
-            version.size != 3 -> false
+            // Incorrect format
+            version == null -> false
             // Major version is invalid
-            version[0] != PRODUCT_NAME_SUPPORTED_SINCE.first -> false
+            version.major != Jellyfin.recommendedVersion.major -> false
```

Comment:
I think the version class supports comparisons, if not we should add that so we don't need to check this manually in this when statement